views/SprinklerView/SprinklerView.tsx

```diff
@@ -0,0 +1,77 @@
+import { NativeStackScreenProps } from '@react-navigation/native-stack';
+import { useMemo } from 'react';
+import React from 'react-native';
+import { Button, Navbar, Typography } from '../../.storybook/stories';
+import { LayoutProvider } from '../../components';
+import { selectDeviceWithId } from '../../redux/devicesSlice';
+import { useAppSelector } from '../../redux/hooks';
+import { RootStackParamList, Sprinkler } from '../../types';
+import { useSendAPIRequest } from '../../hooks';
+import { TurnOnSprinklerSelect } from './TurnOnFanSelect';
+
+type SprinklerViewProps = NativeStackScreenProps<
+  RootStackParamList,
+  'Sprinkler'
+>;
+
+export const SprinklerView = ({ route }: SprinklerViewProps) => {
+  const { location, nodeId, deviceId, botHash, botId } = route.params;
+
+  const sendIoTAPIRequest = useSendAPIRequest();
+
+  const sprinkler = useAppSelector(state =>
+    selectDeviceWithId(state, location, nodeId, deviceId),
+  ) as Sprinkler;
+
+  const sprinklerActionBaseParams = {
+    location,
+    nodeId,
+    deviceId,
+    action: 'set',
+    botHash,
+    botId,
+  };
```

Comment:
It's safer to pass `action` parameters separately in each API request. Also, you don't have to destructure values and move them to the new object.